src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -303,18 +305,22 @@ public String getMonitoringEndpoint(Region region) {
     private List<String> extractResourceIds(List<ResourceTagMapping> resourceTagMappings) {
       List<String> resourceIds = new ArrayList<String>();
       for (ResourceTagMapping resourceTagMapping : resourceTagMappings) {
-        // ARN parsing is based on https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
-        String[] arnArray = resourceTagMapping.getResourceARN().split(":");
-        String resourceId = arnArray[arnArray.length - 1];
-        if (resourceId.contains("/")) {
-          String[] resourceArray = resourceId.split("/", 2);
-          resourceId = resourceArray[resourceArray.length - 1];
-        }
-        resourceIds.add(resourceId);
+        resourceIds.add(extractResourceIdFromArn(resourceTagMapping.getResourceARN()));
       }
       return resourceIds;
     }
     
+    private String extractResourceIdFromArn(String arn) {
+      // ARN parsing is based on https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
+      String[] arnArray = arn.split(":");
+      String resourceId = arnArray[arnArray.length - 1];
+      if (resourceId.contains("/")) {
+        String[] resourceArray = resourceId.split("/", 2);
```

Comment:
I would create a variable with a clear name for the index. What do you think?

Reply:
Here it's not an `index`, it's a `limit`. I would leave it as is.